data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg

```diff
@@ -0,0 +1,101 @@
+#textdomain wesnoth-test
+
+#####
+# API(s) being tested: [firststrike]
+##
+# Actions:
+# Give bob a firststrike ability
+##
+# Expected end state:
+# At time of checking, only side 1 has been struck but side 2 has not been struck
+#####
+{COMMON_KEEP_A_B_UNIT_TEST "firststrike_ability" (
+    [event]
+        name = start
+
+        {VARIABLE is_event 0}
+
+        [modify_unit]
+            [filter]
+                id = bob
+            [/filter]
+            [effect]
+                apply_to = new_ability
+                [abilities]
+                    {TEST_ABILITY_NO_VALUE firststrike () SELF=yes}
+                [/abilities]
+            [/effect]
+        [/modify_unit]
+
+        [modify_unit]
```

Comment:
candidate for macro with [filter][/filter]

Reply:
What do you mean?

Reply:
Looked like code used for multiple tests.

Reply:
It's only for a few tests though, specific to firststrike. Or at least that was my thinking.